bin/node/src/service.rs

```diff
@@ -284,7 +320,74 @@ pub fn new_full(
         justification_rx,
         metrics,
         unit_creation_delay,
+        node_type: NodeType::Validator,
+    };
+    task_manager.spawn_essential_handle().spawn_blocking(
+        "aleph",
+        None,
+        run_aleph_consensus(aleph_config),
+    );
+
+    network_starter.start_network();
+    Ok(task_manager)
+}
+
+pub fn new_nonvalidator(
+    config: Configuration,
+    aleph_config: AlephCli,
+) -> Result<TaskManager, ServiceError> {
+    let sc_service::PartialComponents {
+        client,
+        backend,
+        mut task_manager,
+        import_queue,
+        keystore_container,
+        select_chain,
+        transaction_pool,
+        other: (_, justification_rx, mut telemetry, metrics),
+    } = new_partial(&config)?;
+
+    let (_rpc_handlers, network, network_starter) = setup(
+        config,
+        backend,
+        &keystore_container,
+        import_queue,
+        transaction_pool,
+        &mut task_manager,
+        client.clone(),
+        &mut telemetry,
+    )?;
+
+    let session_period = SessionPeriod(
+        client
+            .runtime_api()
+            .session_period(&BlockId::Number(Zero::zero()))
+            .unwrap(),
+    );
+
+    let millisecs_per_block = MillisecsPerBlock(
+        client
+            .runtime_api()
+            .millisecs_per_block(&BlockId::Number(Zero::zero()))
+            .unwrap(),
+    );
+
+    let unit_creation_delay = aleph_config.unit_creation_delay();
+
+    let aleph_config = AlephConfig {
```

Comment:
I guess this (creation of `AlephConfig`) together with creation of `sessiont_period`, `millisecs_per_block`, `unit_creation_delay` could be moved to `setup(...)`? We would need to also add `select_chain`, `metrics` and `aleph_config` as arguments. WDYT?

Reply:
It makes a mess of the setup fn (and this is something different from setup actually). Imho separate function only for config also does not look good here